### ISSUE TITLE
[syncfusion_flutter_datagrid] not visible column still selectable with arrow key

### ISSUE THREAD

**Issue by Viper-Bit** [syncfusion/flutter-widgets#554]
first column still selectable even with **visible: false** property by left arrow key (or right arrow key in RTL).

reproduce example:
```dart
SfDataGrid(
              source: _employeeDataSource,
              selectionMode: SelectionMode.single,
              navigationMode: GridNavigationMode.cell,
              columnWidthMode: ColumnWidthMode.fill,
              onCurrentCellActivated: (RowColumnIndex newIndex, _) {
                print(
                    'Row: ${newIndex.rowIndex}, Column: ${newIndex.columnIndex}');
              },
              columns: [
                GridColumn(
                  visible: false,
                  columnName: 'id',
                  label: Container(
                    padding: const EdgeInsets.symmetric(horizontal: 16.0),
                    alignment: Alignment.centerRight,
                    child: const Text(
                      'ID',
                      overflow: TextOverflow.ellipsis,
                    ),
                  ),
                ),
                GridColumn(
                  columnName: 'name',
                  label: Container(
                    padding: const EdgeInsets.symmetric(horizontal: 16.0),
                    alignment: Alignment.centerLeft,
                    child: const Text(
                      'Name',
                      overflow: TextOverflow.ellipsis,
                    ),
                  ),
                ),
                GridColumn(
                  columnName: 'designation',
                  label: Container(
                    padding: const EdgeInsets.symmetric(horizontal: 16.0),
                    alignment: Alignment.centerLeft,
                    child: const Text(
                      'Designation',
                      overflow: TextOverflow.ellipsis,
                    ),
                  ),
                ),
                GridColumn(
                  columnName: 'salary',
                  label: Container(
                    padding: const EdgeInsets.symmetric(horizontal: 16.0),
                    alignment: Alignment.centerRight,
                    child: const Text(
                      'Salary',
                      overflow: TextOverflow.ellipsis,
                    ),
                  ),
                ),
              ],
            )
```

**Comment by Tamilarasan-Paranthaman** [comment]
Hi @Viper-Bit,
  
We have considered your reported issue as bug. We will include the changes in our weekly pub release which is expected to be rolled out on 8th February 2022. You can follow up with the below feedback link, 

Feedback link:  https://www.syncfusion.com/feedback/32204/key-navigation-is-performed-for-non-visible-columns-cell-which-is-placed-in-0th

We will let you know once it gets rolled out. We appreciate your patience until then. Please let us know if you require any further assistance.

Regards,
Tamilarasan Paranthaman

**Comment by Tamilarasan-Paranthaman** [comment]
Hi @Viper-Bit ,

Thank you for your patience.
 
We are glad to inform you that the fix has been included in our weekly pub release. Please update DataGrid package to **19.4.50** version.

Please let us know if you need further assistance on this.

Regards,  
Tamilarasan

**Comment by Viper-Bit** [comment]
yea fixed in 19.4.50, thx for support
